src/main/java/org/tron/common/backup/BackupManager.java

```diff
@@ -0,0 +1,150 @@
+package org.tron.common.backup;
+
+import static org.tron.common.backup.BackupManager.BackupStatusEnum.INIT;
+import static org.tron.common.backup.BackupManager.BackupStatusEnum.MASTER;
+import static org.tron.common.backup.BackupManager.BackupStatusEnum.SLAVER;
+import static org.tron.common.net.udp.message.UdpMessageTypeEnum.BACKUP_KEEP_ALIVE;
+
+import io.netty.util.internal.ConcurrentSet;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.Timer;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.tron.common.net.udp.handler.EventHandler;
+import org.tron.common.net.udp.handler.MessageHandler;
+import org.tron.common.net.udp.handler.UdpEvent;
+import org.tron.common.net.udp.message.Message;
+import org.tron.common.net.udp.message.backup.KeepAliveMessage;
+import org.tron.common.net.udp.message.discover.FindNodeMessage;
+import org.tron.common.net.udp.message.discover.NeighborsMessage;
+import org.tron.common.net.udp.message.discover.PingMessage;
+import org.tron.common.net.udp.message.discover.PongMessage;
+import org.tron.common.overlay.discover.node.Node;
+import org.tron.common.overlay.discover.node.NodeHandler;
+import org.tron.core.config.args.Args;
+
+@Component
+public class BackupManager implements EventHandler{
+
+  static final org.slf4j.Logger logger = LoggerFactory.getLogger("BackupManager");
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Explicitly declare the visibility for "logger". ([squid:S2039](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2039))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=951&issues=AWPTJmxcA0k93xG_IQRu&open=AWPTJmxcA0k93xG_IQRu)

[ISSUE_KEY]: AWPTJmxcA0k93xG_IQRu